core/src/mindustry/editor/MapEditorDialog.java

```diff
@@ -604,6 +609,17 @@ private void doInput(){
                 }
             }
         }else{
+            // rip mobile
+            if(Core.input.keyTap(KeyCode.j)){
+                editor.copyData.rotR();
+            }else if(Core.input.keyTap(KeyCode.h)){
+                editor.copyData.rotL();
+            }else if(Core.input.keyTap(KeyCode.x)){
+                editor.copyData.flipX();
+            }else if(Core.input.keyTap(KeyCode.z)){
+                editor.copyData.flipY();
+            }
```

Comment:
Do not hardcode keys, use `Core.input.keyTap(Binding.schematic_flip_x)`, etc.

Use zoom bindings (mouse wheel) to rotate. Add a check for active editor selection [here](https://github.com/Anuken/Mindustry/blob/master/core/src/mindustry/input/DesktopInput.java#L258) so that you can't zoom while an area is selected.